packages/salesforcedx-vscode-apex/src/views/testOutline.ts

```diff
@@ -0,0 +1,465 @@
+        channelService.appendLine('-----------------------------------------');
+        channelService.appendLine(stackTrace);
+        channelService.appendLine(errorMessage);
+        channelService.appendLine('-----------------------------------------');
+        channelService.showChannelOutput();
+      }
+    }
+    if (test.location) {
+      vscode.window.showTextDocument(test.location.uri).then(() => {
+        this.eventsEmitter.emit('Show Highlight', position);
+      });
+    }
+  }
+
+  public async runSingleTest(test: TestNode) {
+    const tmpFolder = this.getTempFolder();
+    const builder = new ReadableApexTestRunCodeActionExecutor(
+      [test.name],
+      false,
+      tmpFolder,
+      this
+    );
+    const commandlet = new SfdxCommandlet(
+      new SfdxWorkspaceChecker(),
+      new EmptyParametersGatherer(),
+      builder
+    );
+    await commandlet.run();
+  }
+
+  public updateSelection(index: vscode.Range | number) {
+    const editor = vscode.window.activeTextEditor;
+    if (editor) {
+      if (index instanceof vscode.Range) {
+        editor.selection = new vscode.Selection(index.start, index.end);
+        editor.revealRange(index); // Show selection
+      } else {
+        const line = editor.document.lineAt(index);
+        const startPos = new vscode.Position(
+          line.lineNumber,
+          line.firstNonWhitespaceCharacterIndex
+        );
+        editor.selection = new vscode.Selection(startPos, line.range.end);
+        editor.revealRange(line.range); // Show selection
+      }
+    }
+  }
+
+  public getTempFolder(): string {
+    return os.tmpdir();
+  }
+
+  public async runApexTests(): Promise<void> {
+    await this.refresh();
+    const tmpFolder = this.getTempFolder();
+    const builder = new ReadableApexTestRunCodeActionExecutor(
+      Array.from(ApexTestOutlineProvider.testStrings.values()),
+      false,
+      tmpFolder,
+      this
+    );
+    const commandlet = new SfdxCommandlet(
+      new SfdxWorkspaceChecker(),
+      new EmptyParametersGatherer(),
+      builder
+    );
+    await commandlet.run();
+  }
+
+  public readJSONFile(folderName: string) {
+    const jsonSummary = this.getJSONFileOutput(folderName);
+    this.UpdateTestsFromJSON(jsonSummary);
+    this.onDidChangeTestData.fire();
+  }
+
+  private getJSONFileOutput(fullFolderName: string): FullTestResult {
+    const files = fs.readdirSync(fullFolderName);
+    let fileName = files[0];
+    for (const file of files) {
+      if (
+        file !== 'test-result-codecoverage.json' &&
+        ospath.extname(file) === '.json' &&
+        file.startsWith('test-result')
+      ) {
+        fileName = file;
+      }
+    }
+    fileName = ospath.join(fullFolderName, fileName);
+    const output = fs.readFileSync(fileName).toString();
+    const jsonSummary = JSON.parse(output) as FullTestResult;
+    return jsonSummary;
+  }
+
+  private UpdateTestsFromJSON(jsonSummary: FullTestResult) {
```

Comment:
`UpdateTestsFromJSON` -> `updateTestsFromJSON`

Method names should be lowerCamelCased.